jms/src/main/scala/akka/stream/alpakka/jms/impl/JmsConnector.scala

```diff
@@ -56,26 +57,40 @@ private[jms] trait JmsConnector[S <: JmsSession] {
   override def preStart(): Unit = {
     // keep two elements since the time between initializing and connected can be very short.
     // always drops the old state, and keeps the most current (two) state(s) in the queue.
-    val pair = Source.queue[InternalConnectionState](2, OverflowStrategy.dropHead).preMaterialize()(this.materializer)
-    connectionStateQueue = pair._1
-    connectionStateSourcePromise.complete(Success(pair._2))
+    val (queue, source) =
+      Source
+        .queue[InternalConnectionState](2, OverflowStrategy.dropHead)
+        .toMat(BroadcastHub.sink(1))(Keep.both)
+        .run()(this.materializer)
+    connectionStateQueue = queue
+    connectionStateSourcePromise.complete(Success(source))
+
+    // add subscription to purge queued connection status events after the configured timeout.
+    val system: ActorSystem = ActorMaterializerHelper.downcast(materializer).system
+    after(jmsSettings.connectionStatusSubscriptionTimeout, system.scheduler) {
```

Comment:
The NullPointerException I menioned before seems to be caused by
```scala
  implicit protected var ec: ExecutionContext = _
````
This executionContext is null at this point and is implicitly passed here...

Reply:
Thanks, fixed now with https://github.com/akka/alpakka/pull/2067/commits/74342089e1b772df16d227463da99e40105c3270